source/common/network/udp_listener_impl.h

```diff
@@ -61,5 +62,18 @@ class UdpListenerImpl : public BaseListenerImpl,
   Event::FileEventPtr file_event_;
 };
 
+class UdpListenerWorkerRouterImpl : public UdpListenerWorkerRouter {
+public:
+  UdpListenerWorkerRouterImpl(uint32_t concurrency);
+
+  void registerWorker(UdpListenerCallbacks& listener) override;
+  void unregisterWorker(UdpListenerCallbacks& listener) override;
+  void deliver(UdpListenerCallbacks& current, UdpRecvData&& data) override;
+
+private:
+  absl::Mutex mutex_;
+  std::vector<UdpListenerCallbacks*> workers_ ABSL_GUARDED_BY(mutex_);
```

Comment:
This vector is not really workers, but listeners. Is it necessary to retain an extra vector here given that there is already one vector of worker instances at ListenerManager. Can we use the existing one or merge them into one?

Reply:
This vector is needed because the referenced object is created/destroyed on the worker thread, and vector is keeping track of which ones currently exist to avoid crashes/races during creation and deletion of the listener on each worker thread.